src/main/java/com/github/andriiyan/sprongtraining/impl/dao/BaseDaoImpl.java

```diff
@@ -16,97 +18,54 @@
 // TODO: 3/20/2022 - any reason in using default access modifier here and in other similar cases?
```

Comment:
@eugenicon didn't get this, could you please explain more ?

Reply:
Many of your -impl classes are not public. No public, protected or private access modifier means that default modifier is used and class is accessible only within same package. I'm not saying it's bad practice - just wondering why you would do that? Is it your conscious decision or just a coincidence?)

Reply:
While creating the app I was thinking that implementation should be hidden and classes should communicate with each other only using public interfaces and shouldn't know anything about its implementation.

So, the main idea that implementation is not visible in the code but visible in the XML Spring configuration and using this Spring feature I could provide implementation of the public interfaces without referencing them in the code.